pages/docs/frontend-modules/development.en-US.mdx

```diff
@@ -13,39 +15,120 @@ yarn
+
+</Callout>
+
+## Using import map overrides
+
+O3 determines which frontend modules to load based on the [import map](/docs/core-concepts#import-map) provided to the app shell. The import map is a JSON object that maps module specifiers to URLs. You can find the import map for your O3 distro by navigating to `/openmrs/spa/importmap.json` in your browser. The [Devtools](https://github.com/openmrs/openmrs-esm-core/tree/main/packages/apps/esm-devtools-app) frotend modules provides an interface for overriding entries in your import map. You can leverage this when doing local development to load copies of your frontend modules from your local dev server. For example, if you fire up a [local development server running the app shell](docs/frontend-modules/development#running-the-app-shell-locally), it's possible to override import map entries for multiple frotend modules from multiple projects at the same time. This is useful when you're working on a feature that spans multiple frontend modules.
+
+### Example: Running the Implementer tools and Primary navigation frontend modules against the app shell
+
+<Steps>
+
+### Run the app shell
+
+```sh
+cd @openmrs/esm-core
+yarn run:shell
+```
+
+This should fire up a dev server on port 8080 (assuming you have no other dev server running).
+
+### Run the implementer tools app
+
+In a different terminal window, run:
+
+```sh
+cd openmrs/esm-core
+yarn start --sources packages/apps/esm-implementer-tools-app --port=7000
+```
+
+This should fire up a dev server on port 8081 (the next available port). To see the actual bundled JavaScript code for the app, navigate to http://localhost:8081/openmrs-esm-implementer-tools-app.js in your browser. This URL is the entry point for the app shell to load the implementer tools app. It's important that the URL points to the correct location of the bundled JavaScript code. Often, you might want to specify a different port number to ensure that the URL points to the correct location than the default port or whatever port the tooling assigns.
+
+<Callout type="info">
+**Port numbers**
+
+By default, the Webpack dev server runs on port 8080. If you're running multiple dev servers, the next available port is assigned to the next dev server. For example, if you're running a dev server on port 8080 and you start another dev server, it will run on port 8081. If you start another dev server, it will run on port 8082, and so on. This is also important to not when setting up import map overrides. If you're running a frontend module using `yarn start` on port 8080, the URL to the bundled JavaScript code will be http://localhost:8081/openmrs-esm-form-builder-app.js. The name of this URL corresponds to the name used in the [browser](https://github.com/openmrs/openmrs-esm-form-builder/blob/main/package.json#L6) entry in the `package.json` file of the frontend module you're running. If you're running a frontend module on a different port, you'll need to adjust the URL accordingly.
```

Comment:
We might need to clarify a bit here, e.g., if you run `openmrs develop` with multiple packages, the app shell will run on 8080 and each package on 8081, 8082, etc. However, if you run each command individually rather than all in one go, you likely have to configure the port yourself. This is one of those things I've never gotten around to fixing but keep meaning to (basically, each server should start on whatever port isn't already assigned).